channels/api.py

```diff
@@ -968,19 +1057,32 @@ def update_post(
             # legacy validation
             raise ValueError("Posts with a url cannot be updated")
 
-        if text:
-            return proxy_post(post.edit(text))
-        if article_content:
-            post.article.content = article_content
-        if cover_image:
-            if hasattr(cover_image, "name"):
-                post.article.cover_image.save(
-                    f"article_image_{post.id}.jpg", cover_image, save=False
-                )
-        elif post.article.cover_image:
-            post.article.cover_image = None
-            post.article.cover_image_small = None
-        post.article.save(update_image=(cover_image is not None))
+        with transaction.atomic():
+            if text:
+                Post.objects.filter(post_id=post_id).update(text=text, edited=True)
+                return proxy_post(post.edit(text))
```

Comment:
Related to my earlier comment: seems like we will also want a helper on `PostProxy` to handle text updates.

This would end up looking something like this:
```python
if text:
  post_proxy = proxy_post(post)
  return post_proxy.edit(text)
```

Reply:
Same thing as me feedback above.